dll/win32/shell32/CUserNotification.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 Hermes Belusca-Maito
+ *
+ * Pass on icon notification messages to the systray implementation
+ * in the currently running shell.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef _USERNOTIFICATION_H_
+#define _USERNOTIFICATION_H_
+
+#undef PlaySound
+
+class CUserNotification :
+    public CComCoClass<CUserNotification, &CLSID_UserNotification>,
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
+    public IUserNotification
+//  public IUserNotification2   // On Vista+
+{
+private:
+    HWND  m_hWorkerWnd;
+    HICON m_hIcon;
+    UINT  m_uTimeout;
+    UINT  m_uInterval;
+    UINT  m_uContinuePoolInterval;
+    UINT  m_cRetryCount;
+    DWORD m_dwInfoFlags;
+    HRESULT m_hRes;
+    IQueryContinue* m_pqc;
```

Comment:
Please use CComPtr. (It will also handle reference counting which you did not do at all now).

Reply:
Unless you manually want to AddRef and Release...

Reply:
Aha ok so it's really still needed. Thanks!

Reply:
Well it appears that since I only store it in Show() for the time of run of this function, then I discard it, it seems that I would have to manually call the CComPtr::Release() function (and also Detach()). It's two calls, and if I just do manually instead the AddRef() and Release() calls I also do two calls, so it seems it would just be simpler to not use CComPtr in that precise case.

@learn-more , any additional advice/comment on that?

Reply:
if you get a COM object, you should always `AddRef` it,
and after you are done, you `Release` it.

If you call `Release` on the CComPtr, it'll zero out the internal field.

Reply:
On Windows (7 and 2003), it appears that neither AddRef nor Release are called on the IQueryContinue* object; only its QueryContinue() function is called.